cmd/ztest/ztest.c

```diff
@@ -624,6 +626,7 @@ usage(boolean_t requested)
 	    "\t[-p pool_name (default: %s)]\n"
 	    "\t[-f dir (default: %s)] file directory for vdev files\n"
 	    "\t[-V] verbose (use multiple times for ever more blather)\n"
+	    "\t[-K] kernel-valid pools only (e.g. no file-type log devices)\n"
```

Comment:
The `zpool import` test cases use pools generated by ztest, in order for those import to always work we can't create pools the kernel modules can't happy.

My preference would be too avoid this issue by allowing the kernel to use L2ARC devices on files.  @ahrens  by chance do you know why this is prohibited in `spa_validate_aux_devs`?  I could see it being useful to be able to use a `/dev/shm/` file as an L2ARC device for testing on Linux.